src/Symfony/Component/Dotenv/Dotenv.php

```diff
@@ -181,7 +177,7 @@ public function populate(array $values, bool $overrideExistingVars = false): voi
         $loadedVars = array_flip(explode(',', $_SERVER['SYMFONY_DOTENV_VARS'] ?? $_ENV['SYMFONY_DOTENV_VARS'] ?? ''));
 
         foreach ($values as $name => $value) {
-            $notHttpName = !str_starts_with($name, 'HTTP_');
+            $notHttpName = !str_starts_with((string) $name, 'HTTP_');
```

Comment:
Not needed. Array keys are either strings or ints, and both are accepted by `str_starts_with()` (as ints can be casted to strings).